google_guest_agent/network/manager/manager.go

```diff
@@ -110,6 +112,57 @@ var (
 	// osinfoGet points to the function to use for getting osInfo.
 	// Primarily used for testing.
 	osinfoGet = osinfo.Get
+
+	// fallbackNetworkManager is the network manager service to be assumed if
+	// none of the known network managers returned true from IsManaging()
+	fallbackNetworkManager Service
+
+	// osRules points to the rules to use for finding relevant ignore rules.
+	// Primarily used for testing.
+	osRules = defaultOSRules
+
+	// defaultOSRules lists the rules for applying interface configurations for primary
+	// and secondary interfaces.
+	defaultOSRules = []osConfigRule{
```

Comment:
why bring back os rules?

Reply:
To avoid changing behavior customers are relying on in ways we don't know about. Specifically thinking of this: https://github.com/GoogleCloudPlatform/guest-agent/issues/401#issuecomment-2235882654 where we are start managing the primary NIC on systems we didn't previously and break a wireguard setup.